pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,50 +329,120 @@ var requiredServices = []string{
-		return nil
+func validateZoneRecords(client *route53.Route53, zone *route53.HostedZone, zoneName string, zonePath *field.Path, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	problematicRecords, err := getSubDomainDNSRecords(client, zone, ic)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(zonePath,
+			errors.Wrapf(err, "could not list record sets for domain %q", zoneName)))
+	}
+
+	if len(problematicRecords) > 0 {
+		detail := fmt.Sprintf(
+			"the zone already has record sets for the domain of the cluster: [%s]",
+			strings.Join(problematicRecords, ", "),
+		)
+		allErrs = append(allErrs, field.Invalid(zonePath, zoneName, detail))
 	}
 
+	return allErrs
+}
+
+func getBaseDomain(session *session.Session, baseDomainPath *field.Path, baseDomainName string) (*route53.HostedZone, field.ErrorList) {
+	baseDomainZone, err := GetPublicZone(session, baseDomainName)
+	if err != nil {
+		return nil, field.ErrorList{
+			field.Invalid(baseDomainPath, baseDomainName, "cannot find base domain"),
+		}
+	}
+	return baseDomainZone, nil
+}
+
+func getHostedZone(client *route53.Route53, hostedZonePath *field.Path, hostedZoneName string) (*route53.GetHostedZoneOutput, field.ErrorList) {
 	// validate that the hosted zone exists
-	hostedZonePath := field.NewPath("aws", "hostedZone")
-	client := route53.New(session)
-	zone, err := client.GetHostedZone(&route53.GetHostedZoneInput{Id: aws.String(ic.AWS.HostedZone)})
+	hostedZoneOutput, err := client.GetHostedZone(&route53.GetHostedZoneInput{Id: aws.String(hostedZoneName)})
 	if err != nil {
-		return field.ErrorList{
-			field.Invalid(hostedZonePath, ic.AWS.HostedZone, "cannot find hosted zone"),
+		return nil, field.ErrorList{
+			field.Invalid(hostedZonePath, hostedZoneName, "cannot find hosted zone"),
 		}
 	}
+	return hostedZoneOutput, nil
+}
 
+func validateHostedZone(hostedZoneOutput *route53.GetHostedZoneOutput, hostedZonePath *field.Path, hostedZoneName string, metadata *Metadata) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	// validate that the hosted zone is associated with the VPC containing the existing subnets for the cluster
 	vpcID, err := metadata.VPC(context.TODO())
 	if err == nil {
-		if !isHostedZoneAssociatedWithVPC(zone, vpcID) {
-			allErrs = append(allErrs, field.Invalid(hostedZonePath, ic.AWS.HostedZone, "hosted zone is not associated with the VPC"))
+		if !isHostedZoneAssociatedWithVPC(hostedZoneOutput, vpcID) {
+			allErrs = append(allErrs, field.Invalid(hostedZonePath, hostedZoneName, "hosted zone is not associated with the VPC"))
 		}
 	} else {
-		allErrs = append(allErrs, field.Invalid(hostedZonePath, ic.AWS.HostedZone, "no VPC found"))
+		allErrs = append(allErrs, field.Invalid(hostedZonePath, hostedZoneName, "no VPC found"))
 	}
 
+	return allErrs
+}
+
+func getSubDomainDNSRecords(client *route53.Route53, hostedZone *route53.HostedZone, ic *types.InstallConfig) ([]string, error) {
+	var err error
```

Comment:
Don't declare this variable here. Declare it where it is used via `:=`.

Reply:
Fixed.